### PR TITLE
BaSP-TG: Time sheets methods (Create, Edit and Get)

### PR DESCRIPTION
**Pull request by Licamus**
We worked together with @jivescovo, and we created the methods Create Time Sheet, Edit Time Sheet and Get one Time Sheet by ID.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -1,21 +1,22 @@
...
 
 app.use(express.json());
+app.use('/timeSheet', timeSheetRouter);
 
 app.get('/', async (req, res) => {
   res.send('Hello World!');
 });
 
-app.get('/admins', (req, res) => {
+app.get('/timeSheets', (req, res) => {
```
Comment (ignacioaristo) [inline src/index.js:17]: We don't have to handle this logic here.

File: src/index.js [hunk 2]
```diff
@@ -1,21 +1,22 @@
 // use "import" to import libraries
 import express from 'express';
 
-// use "require" to import JSON files
-const admins = require('./data/admins.json');
+const timeSheets = require('./data/time-sheets.json');
```
Comment (ignacioaristo) [inline src/index.js:4]: We don't need this here. We handle this logic in the `resources` folder.

File: src/resources/time-sheets.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+const express = require('express');
+const fs = require('fs');
+const timeSheets = require('../data/time-sheets.json');
+
+const router = express.Router();
+
+// Add time sheet
+router.post('/addTimeSheet', (req, res) => {
```
Comment (ignacioaristo) [inline src/resources/time-sheets.js:8]: We could name this path with just `add` in order to not be redundant with the path

File: src/resources/time-sheets.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+  && !(timeSheets.find((timeSheet) => JSON.stringify(timeSheet.id)
+  === JSON.stringify(newTimeSheet.id)))) {
+    if (Object.keys(newTimeSheet)[1] === 'description'
+    && Object.keys(newTimeSheet)[2] === 'date' && Object.keys(newTimeSheet)[3] === 'task') {
+      timeSheets.push(newTimeSheet);
+      fs.writeFile('src/data/time-sheets.json', JSON.stringify(timeSheets), (err) => {
+        if (err) {
+          res.send('Problem when adding time sheet');
+        } else {
+          res.send('timeSheet created');
```
Comment (ignacioaristo) [inline src/resources/time-sheets.js:20]: We could add the timeSheet created, something like `timeSheet created: ${newTimeSheet}`

File: src/resources/time-sheets.js [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+    }
+  } else if (JSON.stringify(newTimeSheet) === '{}') {
+    res.send('Cannot add empty time sheet');
+  } else {
+    res.send('Id cannot be repeated or id property name is incorrect');
+  }
+});
+
+// Get time sheet method
+router.get('/getTimeSheetById/:id', (req, res) => {
```
Comment (ignacioaristo) [inline src/resources/time-sheets.js:34]: We could name this path `getById/:id` in order to not be redoundant.

File: src/resources/time-sheets.js [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+        timeSheet.date = updateTimeSheet.date ? updateTimeSheet.date : timeSheet.date; // eslint-disable-line
+        timeSheet.task = updateTimeSheet.task ? updateTimeSheet.task : timeSheet.task; // eslint-disable-line
+        fs.writeFile('src/data/time-sheets.json', JSON.stringify(timeSheets), (err) => {
+          if (err) {
+            res.send('Problem when adding time sheet');
+          } else {
+            res.send('time sheet created');
+          }
+        });
+        res.json({ msg: 'time sheet updated', timeSheet });
```
Comment (ignacioaristo) [inline src/resources/time-sheets.js:63]: I like this form of showing the response, maybe we could use it in the rest of the endpoints.

**Review by ignacioaristo — CHANGES_REQUESTED** [review]

File: src/index.js (hunk 1 above)
Comment (Licamus) [reply src/index.js]: Done!

File: src/index.js (hunk 2 above)
Comment (Licamus) [reply src/index.js]: Done!

File: src/resources/time-sheets.js (hunk 3 above)
Comment (Licamus) [reply src/resources/time-sheets.js]: Done!

File: src/resources/time-sheets.js (hunk 4 above)
Comment (Licamus) [reply src/resources/time-sheets.js]: Done!

File: src/resources/time-sheets.js [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+  === timeSheetId);
+  if (foundTimeSheet) {
+    res.send(foundTimeSheet);
+  } else {
+    res.send('time sheet not found');
+  }
+});
+
+// Edit time Sheet method
+router.put('/modTimeSheet/:id', (req, res) => {
```
[the remark this answers is not in the document]
Comment (Licamus) [reply src/resources/time-sheets.js]: Done!

File: src/resources/time-sheets.js (hunk 6 above)
Comment (Licamus) [reply src/resources/time-sheets.js:63]: Done!

**Review by ignacioaristo — APPROVED** [review]

**Pull request merged**